### PR TITLE
#72/Create controller and db adapter to update user info

### LINKED ISSUE #72: Create PUT/PATCH controller to update certain information of the user

### LINKED ISSUE #73: Create db adapter to update certain user information

### PR DESCRIPTION
**Pull request by gbudau**
This PR adds one endpoint:

PATCH [/api/v1/users](http://localhost:3000/api/#/users/UserController_updateCurrentUser) -> update the info of the currently authenticated user

image

close #72 
close #73

### CODE REVIEW

File: transcendence-app/src/user/dto/update-user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { IsEmail, IsOptional } from 'class-validator';
+
+export class UpdateUserDto {
+  @IsEmail()
+  @IsOptional()
+  email?: string;
```
Comment (gbudau) [inline transcendence-app/src/user/dto/update-user.dto.ts:6]: We could add more options to this DTO when we add more fields to the user table.

Comment (Damrod) [reply transcendence-app/src/user/dto/update-user.dto.ts:6]: Why is `UpdateUserDto` better than this `Partial<User>`? They seem largely overlapping in function and content

**Review by Damrod — APPROVED** [review]
Lgtm!

File: transcendence-app/src/user/dto/update-user.dto.ts (hunk 1 above)
Comment (gbudau) [reply transcendence-app/src/user/dto/update-user.dto.ts:6]: `Partial<User>` allows the users to update all the fields from the `Users` table. Including `avatarId`, `createdAt`, and any column we might add in the future.
This class limits the fields the users can update.

**Review by fizcamposancos — COMMENTED** [review]
LGTM

**Pull request merged**